pyartcd/pyartcd/redis.py

```diff
@@ -85,3 +86,51 @@ async def delete_key(conn: aioredis.commands.Redis, key: str) -> int:
     res = await conn.delete(key)
     logger.debug('Key %s %s', key, 'deleted' if res else 'not found')
     return res
+
+
+@handle_connection
+async def list_push(conn: aioredis.commands.Redis, key: str, value: str):
+    """
+    Push to a list in Redis, like a queue (FIFO). Docs: https://redis.io/docs/data-types/lists/
+    """
+
+    logger.debug('Pushing value %s to list %s', value, key)
+    await conn.lpush(key, value)
+
+
+@handle_connection
+async def list_pop(conn: aioredis.commands.Redis, key: str) -> (str, None):
+    """
+    Pop a value from a list, like a queue (FIFO). Docs: https://redis.io/docs/data-types/lists/
+    :return: A string if a value is present, else None
+    """
+
+    value = await conn.rpop(key)
+    logger.debug('List %s has value %s', key, value)
+    return value
+
+
+@handle_connection
+async def list_pop_all(conn: aioredis.commands.Redis, key: str) -> list:
+    """
+    Get all values from the Redis list
+
+    :return: List of values from the key
+    """
+    values = []
+    while True:
```

Comment:
I think there should be some function in redis to specifically read/write lists. This looks quite inefficient

Reply:
Found the read one! thank you! https://github.com/openshift-eng/art-tools/pull/90/commits/a6b9789f3280c3d5f2a8a507226a921d1f223a3f

Couldn't find the bulk write one though